### ISSUE TITLE
pixi 0.27.1 fails to parse project manifest with *xx* pattern version

### ISSUE THREAD

**Issue by RenaudLN** [prefix-dev/pixi#1782]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

Add a dependency with special pattern like the following to pyproject.toml

```
[tool.pixi.dependencies]
jaxlib = "*cpu*"
```

### Issue description

Pixi commands fail with the following error message:
![image](https://github.com/user-attachments/assets/e40488cb-c46c-4c9d-a804-16f7f14d5849)

if I rollback to 0.27.0 it works as expected

### Expected behavior

Pixi should be able to parse these pattern versions, as it did in prior versions.

**Comment by baszalmstra** [comment]
I will have to investigate but I think in the previous version this would also not parse properly. We expect a version string not a matchspec. I think what you want to achieve is:

```
jaxlib = { build = "*cpu*" }
```

E.g. select any version that has `cpu` anywhere in the build string.

**Comment by baszalmstra** [comment]
I made an attempt at improving the error message: https://github.com/prefix-dev/pixi/pull/1786

**Comment by RenaudLN** [comment]
It was definitely working with 0.27.0 but thanks, the 'build' fix works with 0.27.1!
![image](https://github.com/user-attachments/assets/1eec4c95-f6db-4eac-ab9c-61027e6eb282)

And thanks for the error message update!

**Comment by baszalmstra** [comment]
Closing this now that #1786 has been merged.
